File: CWE-807/example_1.php
```
	$auth = $_COOKIES['authenticated'];
	if (! $auth) {
		if (AuthenticateUser($_POST['user'], $_POST['password']) == "success") {
			// save the cookie to send out in future responses
			setcookie("authenticated", "1", time()+60*60*2);
		}
		else {
			ShowLoginScreen();
			die("\n");
		}
	}
	DisplayMedicalHistory($_POST['patient_ID']);
```


//Result: 
 // Vulnerable (01)
//
//
//
////Issue: CWE-807 - Reliance on Untrusted Inputs for Security Decisions
//
//
//
//URL: https://cwe-api.mitre.org/api/v1/cwe/weakness/807
//

//Source: CWE Mitre
//

//Extra context: 
// The following code could be for a medical records application. It performs authentication by checking if a cookie has been set.
// The programmer expects that the AuthenticateUser() check will always be applied, and the "authenticated" cookie will only be set when authentication succeeds. The programmer even diligently specifies a 2-hour expiration for the cookie.
// However, the attacker can set the "authenticated" cookie to a non-zero value such as 1. As a result, the $auth variable is 1, and the AuthenticateUser() check is not even performed. The attacker has bypassed the authentication.

File: CWE-798/example_1.java
```
	int VerifyAdmin(String password) {
		if (!password.equals("Mew!")) {
			return(0)
		}
		//Diagnostic Mode
		return(1);
	}
```


//Result: 
 // Vulnerable (01)
//
//
//
////Issue: CWE-798 - Use of Hard-coded Credentials
//
//
//
//URL: https://cwe-api.mitre.org/api/v1/cwe/weakness/798
//

//Source: CWE Mitre
//

//Extra context: 
// The following code is an example of an internal hard-coded password in the back-end:
// Every instance of this program can be placed into diagnostic mode with the same password. Even worse is the fact that if this program is distributed as a binary-only distribution, it is very difficult to change that password or disable this "functionality."